### PR TITLE
Update TapLocation screen

### PR DESCRIPTION
**Pull request by shubham7109**
Related to issue: `kotlin/issues/4115`

### Description:

Update the TapLocationScreen to place the `CalloutOptionsBox` in a dismissable BottomSheet

### Summary of changes:

- Added a bottom sheet container
- Updated the calloutVisibility to be set to `true` by default
- Optimized the CalloutOptionsBox layout to take up less space 

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [x] No

### CODE REVIEW

File: microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/TapLocationScreen.kt [hunk 1]
```diff
@@ -50,46 +57,58 @@ import kotlin.math.roundToInt
...
+        scaffoldState = bottomSheetScaffoldState,
+    ) { paddingValues ->
         MapView(
-            modifier = Modifier
-                .fillMaxSize(),
+            modifier = Modifier.fillMaxSize().padding(paddingValues),
             arcGISMap = viewModel.arcGISMap,
             graphicsOverlays = remember { listOf(viewModel.tapLocationGraphicsOverlay) },
             onSingleTapConfirmed = viewModel::setMapPoint,
             onLongPress = { viewModel.clearMapPoint() },
```
Comment (puneet-pdx) [inline microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/TapLocationScreen.kt:102]: We should remove the onLongPress here as well to clear the TapPoint and add a button ?

**Comment by shubham7109** [comment]
@puneet-pdx  Ready for your review.

**Review by puneet-pdx — APPROVED** [review]

File: microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/TapLocationScreen.kt [hunk 2]
```diff
@@ -173,3 +162,42 @@ fun CalloutOptions(
...
+    offsetName: String,
+    onOffsetSelected: (Float) -> Unit,
+) {
+    val offsetsItems = listOf(-100.0, -50.0, -25.0, 0.0, 25.0, 50.0, 100.0)
+    var isXOffsetsExpanded by rememberSaveable { mutableStateOf(false) }
+    ExposedDropdownMenuBox(
+        modifier = modifier,
+        expanded = isXOffsetsExpanded,
+        onExpandedChange = { isXOffsetsExpanded = !isXOffsetsExpanded }
+    ) {
```
Comment (shubham7109) [inline microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/TapLocationScreen.kt]: @puneet-pdx @eri9000 I replaced the `OutlineTextField` for Offsets with an `ExposedDropdownMenuBox` to allow for a dropdown selection of Offsets. Feel free to add any feedback if needed. 

Screenshot 2024-06-14 at 4 17 32 PM

Comment (shubham7109) [reply microapps/MapViewCalloutApp/app/src/main/java/com/arcgismaps/toolkit/mapviewcalloutapp/screens/TapLocationScreen.kt]: @puneet-pdx @eri9000 As discussed in scum, reverted to use OutlinedTextField, and added the Offsets unit (px) as part of the text field label.

**Review by eri9000 — APPROVED** [review]

**Pull request merged**